SWIPs/swip-data-types-descriptions.md

```diff
@@ -0,0 +1,77 @@
+---
+SWIP: <to be assigned>
+title: Type definitions for protocol messages
+author: Louis Holbrook @nolash <dev@holbrook.no>
+status: Draft
+type: General
+created: 2019-08-02
+---
+
+## abstract
+
+This SWIP defines pseudo data types for use in Swarm protocol specifications.
+
+## motivation
+
+To describe protocols in our specifications, we are in need of a strongly defined way of describing data types and their serializations to RLP.
+
+RLP does not have "types" as such, just self-describing data units whose length is serially interpreted. Hence it is for example not possible to define a number value range merely in terms of RLP constructs.
```

Comment:
Essentially we want to define a more well-defined subset of RLP? How does this look in ETH1 (or eth2 for ssz)?

Reply:
And how does this generalize beyond RLP? I.e. how would it be compatible with libp2p and some other serialization?

Reply:
You can still use RLP over libp2p no?

Anyway, do you mean that we should have a reference document that defines serializations of our declaration format to relevant wire streams. Would there be others than RLP and perhaps protobuf initially?